### PR TITLE
[tokio] migrate to 0.2.0-alpha.6

### PR DESCRIPTION
**Pull request by bmwill**
Migrate to using tokio 0.2.0-alpha.6.

### CODE REVIEW

File: network/netcore/src/compat.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
+use pin_project::pin_project;
```
Comment (bothra90) [inline network/netcore/src/compat.rs]: Can you add comments on what the `Compat` struct is useful for?

Comment (bmwill) [reply network/netcore/src/compat.rs]: Added some brief comments. I'm hoping that the community will end up standardizing on a set of traits but in the mean time we can use a compat layer.

File: network/netcore/src/compat.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    pin::Pin,
+    task::{self, Poll},
+};
+
+/// `Compat` provides a compatibility shim between the `AsyncRead`/`AsyncWrite` traits provided by
+/// the `futures` library and those provided by the `tokio` library since they are different and
+/// incompatible with one another.
+#[pin_project]
+#[derive(Copy, Clone, Debug)]
+pub struct Compat<T> {
```
Comment (bothra90) [inline network/netcore/src/compat.rs]: nit: Can we give a slightly more descriptive name, like `IoCompat`?

File: network/src/protocols/rpc/mod.rs [hunk 3]
```diff
@@ -297,12 +297,15 @@ async fn handle_outbound_rpc<TSubstream>(
...
                 .timeout(timeout)
-                .compat()
-                // Convert tokio timeout::Error to RpcError
-                .map_err(Into::<RpcError>::into);
+                .map_err(Into::<RpcError>::into)
+                .map(|r| match r {
+                    Ok(Ok(x)) => Ok(x),
+                    Ok(Err(e)) => Err(e),
+                    Err(e) => Err(e),
+                })
```
Comment (bothra90) [inline network/src/protocols/rpc/mod.rs]: Is this due to changes in upstream timeout future? Also, should we create a function to collapse a `Result<Result<T, E>, E>` into a `Result<T, E>` and use that at various timeout callsites?

**Comment by phlip9** [comment]
`futures-semaphore` changes LGTM; thanks for taking care of this!

**Pull request closed without merging**